src/overlays/actors/ovl_Bg_Ganon_Otyuka/z_bg_ganon_otyuka.c

```diff
@@ -31,18 +32,360 @@ const ActorInit Bg_Ganon_Otyuka_InitVars = {
+        Math_SmoothDownscaleMaxF(&this->flashYScale, 1.0f, 0.25f);
+        if (this->flashYScale == 0.0f) {
+            this->flashState = FLASH_NONE;
+        }
+    }
+    if (this->dropTimer == 0) {
+        this->flashYScale = 0.0f;
+        Math_SmoothScaleMaxF(&this->dyna.actor.posRot.pos.y, -1000.0f, 1.0f, this->dyna.actor.speedXZ);
+        Math_SmoothScaleMaxF(&this->dyna.actor.speedXZ, 100.0f, 1.0f, 2.0f);
+        if (!(this->unwalledSides & SIDE_EAST)) {
+            this->dyna.actor.shape.rot.z -= (s16)(this->dyna.actor.speedXZ * 30.0f);
+        }
+        if (!(this->unwalledSides & SIDE_WEST)) {
+            this->dyna.actor.shape.rot.z += (s16)(this->dyna.actor.speedXZ * 30.0f);
+        }
+        if (!(this->unwalledSides & SIDE_SOUTH)) {
+            this->dyna.actor.shape.rot.x += (s16)(this->dyna.actor.speedXZ * 30.0f);
+        }
+        if (!(this->unwalledSides & SIDE_NORTH)) {
+            this->dyna.actor.shape.rot.x -= (s16)(this->dyna.actor.speedXZ * 30.0f);
+        }
+        if (this->dyna.actor.posRot.pos.y < -750.0f) {
+            if (player->actor.posRot.pos.y < -400.0f) {
+                accel.x = accel.z = 0.0f;
+                accel.y = 0.1f;
+                velocity.x = velocity.y = velocity.z = 0.0f;
+
+                for (i = 0; i < 30; i++) {
+                    pos.x = Math_Rand_CenteredFloat(150.0f) + this->dyna.actor.posRot.pos.x;
+                    pos.y = Math_Rand_ZeroFloat(60.0f) + -750.0f;
+                    pos.z = Math_Rand_CenteredFloat(150.0f) + this->dyna.actor.posRot.pos.z;
+                    func_8002836C(globalCtx, &pos, &velocity, &accel, &sDustPrimColor, &sDustEnvColor,
+                                  (s16)Math_Rand_ZeroFloat(100.0f) + 250, 5, (s16)Math_Rand_ZeroFloat(5.0f) + 15);
+                }
+
+                func_80033DB8(globalCtx, 10, 15);
+                Audio_PlaySoundAtPosition(globalCtx, &this->dyna.actor.posRot.pos, 0x28, 0x2839);
+            }
+            Actor_Kill(&this->dyna.actor);
+        }
+    } else {
+        if (this->dropTimer == 1) {
+            Audio_PlaySoundGeneral(NA_SE_EV_STONEDOOR_STOP, &this->dyna.actor.projectedPos, 4, &D_801333E0, &D_801333E0,
+                                   &D_801333E8);
+        } else {
+            Audio_PlaySoundGeneral(NA_SE_EV_BLOCKSINK - SFX_FLAG, &this->dyna.actor.projectedPos, 4, &D_801333E0,
+                                   &D_801333E0, &D_801333E8);
+        }
+        Math_SmoothScaleMaxF(&this->dyna.actor.posRot.pos.y, -1000.0f, 1.0f, this->dyna.actor.speedXZ);
+        Math_SmoothScaleMaxF(&this->dyna.actor.speedXZ, 100.0f, 1.0f, 0.1f);
+    }
+    osSyncPrintf("MODE DOWN END\n");
+}
+
+void BgGanonOtyuka_DoNothing(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void BgGanonOtyuka_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgGanonOtyuka* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+    this->flashTimer++;
+    DECR(this->dropTimer);
```

Comment:
```suggestion
    if (this->dropTimer != 0) {
        this->dropTimer--;
    }
```

Reply:
Question, how come here we wouldn't use `DECR` here? Is it more for decrementing timers in if statements?

Reply:
Yeah it's intended for use inline in conditions. We avoid using it otherwise because some decrements towards 0 are done with `> 0` checks and such, so we can't use it consistently.